Manifest validator package (#24)
LedgerHQ/wallet-api: packages/wallet-api-manifest-validator/README.md

<p align="center">
 <img src="https://user-images.githubusercontent.com/9203826/154288895-670f5c23-81a1-4307-a080-1af83f7f8356.svg" align="center" alt="Ledger" />
 <h2 align="center">@ledgerhq/wallet-api-manifest-validator</h2>
 <p align="center">Verify your Live App manifest inside <a href="https://www.ledger.com/ledger-live">Ledger Live</a></p>
</p>
  <p align="center">
    <a href="https://www.npmjs.com/package/">
      <img src="https://img.shields.io/npm/v/" />
    </a>
    <a href="https://discord.gg/y6nZhxv2bC">
      <img alt="Discord" src="https://img.shields.io/discord/885256081289379850?color=1C1CE1&label=Ledger%20%7C%20Discord%20%F0%9F%91%8B%20&style=flat-square" />
    </a>
    <a href="https://opensource.org/licenses/MIT">
      <img alt="lICENCE" src="https://img.shields.io/badge/License-MIT-yellow.svg" />
    </a>

  </p>

  <p align="center">
    <a href="https://developers.ledger.com/docs/dapp/manifest">About the manifest</a>
    ·
    <a href="https://github.com/LedgerHQ/wallet-api/issues/new/choose">Report Bug</a>
    ·
    <a href="https://github.com/LedgerHQ/wallet-api/issues/new/choose">Request Feature</a>
  </p>
</p>

## Quick intro

> The manifest Validator is a typescript package that checks if your `manifest.json` file meets the requirements for Ledger Live App manifest submission.

Reviewer: We should probably remove all Ledger Live references
```suggestion
The manifest Validator is a typescript package that checks if your `manifest.json` file meets the requirements for a Wallet App manifest submission.
```

To achieve this, the package uses [JSON Schema](https://json-schema.org/), a vocabulary for annotating and validating JSON documents.

> In top of that, we us the [AJV](https://ajv.js.org/) and [AJV-errors](https://www.npmjs.com/package/ajv-errors) libraries which provides more possibilities (please refer to [AJV Github](https://github.com/ajv-validator/ajv) to know which version of JSON Schema we use).

Reviewer: small typo
```suggestion
In top of that, we use the [AJV](https://ajv.js.org/) and [AJV-errors](https://www.npmjs.com/package/ajv-errors) libraries which provides more possibilities (please refer to [AJV Github](https://github.com/ajv-validator/ajv) to know which version of JSON Schema we use).
```

## Installation

```bash
npm i @ledgerhq/wallet-api-manifest-validator

```

> _Can also be used globaly :_

Reviewer 2: ```suggestion
_Can also be used globaly:_
```

```bash
npm i -g @ledgerhq/wallet-api-manifest-validator
```

## How to use

#### Import

```typescript
import { validateManifest } from "@ledgerhq/wallet-api-manifest-validator";
import MyjsonManifestFile from "./manifest.json";

const result: boolean = validateManifest(MyjsonManifestFile);
```

#### CLI

> Once you have installed it, use the command :

Reviewer 2: ```suggestion
Once you have installed it, use the command:
```

```bash
npm run validate <*fileOrDirectoryName*>
```

> _Or if you have installed it in global :_

Reviewer 2: ```suggestion
_Or if you have installed it in global:_
```

| options  | Object | More details bellow                                           |

**Options**

|   Options   |  Type   | Description                                                                                 |
| :---------: | :-----: | ------------------------------------------------------------------------------------------- |
|   details   | boolean | describe errors when they occur                                                             |
| enableState | boolean | result description (e.g. show in console "The JSON file does not correspond to the schema") |
|  fileName   | string  | file name, practical to validate multiple files via multiple calls                          |

Imported in source code:

```typescript
validateManifest(MyjsonManifestFile, { details, enableState, fileName });
```

_Cli command don't allow you to use the fileName option._
_--throwError option is only available on cli._

Using CLI:

```bash
npm run validate *fileOrDirectoryName* --details --enableState --throwError
```                                                                                                                                          

</br>
</br>
</br>

> more infos about the manifest here :

Reviewer 2: ```suggestion
more infos about the manifest here:
```

> https://developers.ledger.com/docs/dapp/manifest/

Reviewer 2: You should also link to the non dapp manifest page -> https://developers.ledger.com/docs/non-dapp/tutorial/2-manifest/

At least until to doc is reworked and we have only one page to document manifests (probably directly in this repo in the Docusaurus doc once it will be setup)

```suggestion
- https://developers.ledger.com/docs/dapp/manifest/
- https://developers.ledger.com/docs/non-dapp/tutorial/2-manifest/
```
